Formula/r/rust.rb

```diff
@@ -4,14 +4,24 @@ class Rust < Formula
   license any_of: ["Apache-2.0", "MIT"]
 
   stable do
-    url "https://static.rust-lang.org/dist/rustc-1.71.1-src.tar.gz"
-    sha256 "6fa90d50d1d529a75f6cc349784de57d7ec0ba2419b09bde7d335c25bd4e472e"
+    url "https://static.rust-lang.org/dist/rustc-1.72.0-src.tar.gz"
+    sha256 "ea9d61bbb51d76b6ea681156f69f0e0596b59722f04414b01c6e100b4b5be3a1"
 
     # From https://github.com/rust-lang/rust/tree/#{version}/src/tools
     resource "cargo" do
       url "https://github.com/rust-lang/cargo.git",
-          tag:      "0.72.2",
-          revision: "1a737af0c83b28c1f249b821a76a19c82696b05a"
+          tag:      "0.73.1",
+          revision: "26bba48309d080ef3a3a00053f4e1dc879ef1de9"
+    end
+
+    # Fix for a compiler issue that may cause some builds to spin forever.
+    # Remove on 1.73.0 release.
+    # https://github.com/rust-lang/rust/pull/114948
+    # https://github.com/rust-lang/rust/issues/115297
+    # https://github.com/kpcyrd/sh4d0wup/issues/12
+    patch do
+      url "https://github.com/rust-lang/rust/commit/0f7f6b70617fbcda9f73755fa9b560bfb0a588eb.patch?full_index=1"
+      sha256 "549f446612bef10a1a06a967f61655b59c1b6895d762d764ca89caf65c114fe9"
     end
```

Comment:
Confirmed that this fixed the `sh4d0wup` issue. We should be good for another CI run.